"""Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT."""

from __future__ import annotations
from attio.errors import SDKError
from attio.models import (
    put_v2_objects_object_recordsop as models_put_v2_objects_object_recordsop,
)
from attio.types import BaseModel
from dataclasses import dataclass, field
import httpx
from typing import Optional


class PutV2ObjectsObjectRecordsNotFoundErrorData(BaseModel):
    status_code: float

    type: models_put_v2_objects_object_recordsop.PutV2ObjectsObjectRecordsNotFoundType

    code: models_put_v2_objects_object_recordsop.PutV2ObjectsObjectRecordsNotFoundCode

    message: str


@dataclass(frozen=True)
class PutV2ObjectsObjectRecordsNotFoundError(SDKError):
    r"""Not Found"""

    data: PutV2ObjectsObjectRecordsNotFoundErrorData = field(hash=False)

    def __init__(
        self,
        data: PutV2ObjectsObjectRecordsNotFoundErrorData,
        raw_response: httpx.Response,
        body: Optional[str] = None,
    ):
        fallback = body or raw_response.text
        message = str(data.message) or fallback
        super().__init__(message, raw_response, body)
        object.__setattr__(self, "data", data)


class PutV2ObjectsObjectRecordsValueNotFoundErrorData(BaseModel):
    status_code: float

    type: models_put_v2_objects_object_recordsop.PutV2ObjectsObjectRecordsBadRequestType

    code: models_put_v2_objects_object_recordsop.PutV2ObjectsObjectRecordsCodeValueNotFound

    message: str


@dataclass(frozen=True)
class PutV2ObjectsObjectRecordsValueNotFoundError(SDKError):
    r"""Bad Request"""

    data: PutV2ObjectsObjectRecordsValueNotFoundErrorData = field(hash=False)

    def __init__(
        self,
        data: PutV2ObjectsObjectRecordsValueNotFoundErrorData,
        raw_response: httpx.Response,
        body: Optional[str] = None,
    ):
        fallback = body or raw_response.text
        message = str(data.message) or fallback
        super().__init__(message, raw_response, body)
        object.__setattr__(self, "data", data)
